File: CSharp/SpecStudioExamplesTestsCSharp/production/IDValue.cs
namespace production
{
    public class IDValue
    {
        public IDForm ID { get; }
        public int Value { get; }

        public IDValue(IDForm iD, int value)
        {
            ID = iD;
            Value = value;
        }

        public class Builder
        {
            private IDForm _iD;
            private int _value;

            public Builder ID(IDForm value) { _iD = value; return this; }
            public Builder Value(int value) { _value = value; return this; }

            public IDValue Build() =>
                new IDValue(_iD, _value);
        }
    }
}

File: Java/SpectableJavaExamplesTest/src/main/java/production/IDValue.java
package production;

import java.util.Objects;

public class IDValue {
    public final IDForm iD;
    public final int value;

    public IDValue(IDForm iD, int value) {
        this.iD = iD;
        this.value = value;
    }

    public static class Builder {
        private IDForm iD;
        private int value;

        public Builder iD(IDForm iD) { this.iD = iD; return this; }
        public Builder value(int value) { this.value = value; return this; }

        public IDValue build() {
            return new IDValue(iD, value);
        }
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof IDValue)) return false;
        IDValue that = (IDValue) o;
        return Objects.equals(iD, that.iD)
            && Objects.equals(value, that.value);
    }

    @Override
    public int hashCode() {
        return Objects.hash(iD, value);
    }

    @Override
    public String toString() {
        return "IDValue{" + "ID=" + iD + ", " + "Value=" + value + "}";
    }
}
